horace_core/sqw/@Experiment/Experiment.m

```diff
@@ -429,6 +489,84 @@
+            %          added. this could be a unique_references_container
+            %          or unique_objects_container with baseclass 'type'
+            %          or an object which 'isa(,type)'
+            % - type:  string naming the baseclass to which val should
+            %          belong
+            %
+            % Output
+            % ------
+            % - obj:   the modified Experiment object where the value of
+            %          the field named by the calling form of val (using
+            %          inputname(val#argnum) is modified by adding val if
+            %          val is not a unique_references-container or setting
+            %          it if it is a unique_references_container
+
+            % get the name of the calling variable which enters as 'val'
+            field = [inputname(2) '_'];
+
+            if isa(val, 'unique_references_container') && ...
+                   strcmp(val.stored_baseclass,type)
+                % if size is right, overwrite default compressed container
+                if val.n_runs == obj.n_runs
+                    obj.(field) = val;
+                else
+                    error('HORACE:Experiment:invalid_argument', ...
+                          'input %d size must match number of runs',obj.n_runs);
+                end
+
+
+            elseif ( isa(val, type) &&                       ...
+                     numel(val) == obj.n_runs )              ...
+                   ||                                        ...                   
+                   ( isa(val, 'unique_objects_container') && ...
+                     val.n_runs == obj.n_runs             && ...
+                     strcmp(val.baseclass, type) )           ...
+                   ||                                        ...
+                   ( iscell(val)                          && ...
+                     numel(val) == obj.n_runs             && ...
+                     isa(val{1}, type) )
+
+                % add to default compressed container
+                obj.(field) = obj.(field).add(val);
+
+            elseif ( isa(val, type) &&                       ...
+                     numel(val) == 1 )                       ...
+                   ||                                        ...
+                   ( iscell(val)                          && ...
+                     numel(val) == 1                      && ...
+                     isa(val{1}, type) )
+                % assume we're adding n_runs identical copies
+                % 
+                % add to default compressed container
+                %{
+                for i=1:obj.n_runs
+                    obj.(field) = obj.(field).add(val);
+                end
+                %}
+                obj.(field) = obj.(field).add_copies_(val, obj.n_runs);
+
+            elseif isempty(val)
+                % do nothing, leave default compressed container empty
+                ;
```

Comment:
Would probably have this as the first if and explicit `return`

Reply:
the order goes from most likely and useful combination down to minor alternatives